python/src/main/java/org/apache/zeppelin/python/PythonInterpreter.java

```diff
@@ -0,0 +1,236 @@
+/*
+* Licensed to the Apache Software Foundation (ASF) under one or more
+* contributor license agreements.  See the NOTICE file distributed with
+* this work for additional information regarding copyright ownership.
+* The ASF licenses this file to You under the Apache License, Version 2.0
+* (the "License"); you may not use this file except in compliance with
+* the License.  You may obtain a copy of the License at
+*
+*  http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+package org.apache.zeppelin.python;
+
+import org.apache.zeppelin.interpreter.Interpreter;
+import org.apache.zeppelin.interpreter.InterpreterContext;
+import org.apache.zeppelin.interpreter.InterpreterPropertyBuilder;
+import org.apache.zeppelin.interpreter.InterpreterResult;
+import org.apache.zeppelin.interpreter.InterpreterResult.Code;
+import org.apache.zeppelin.scheduler.Job;
+import org.apache.zeppelin.scheduler.Scheduler;
+import org.apache.zeppelin.scheduler.SchedulerFactory;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.*;
+import java.lang.reflect.Field;
+import java.util.Collection;
+import java.util.List;
+import java.util.Properties;
+
+/**
+ * Python interpreter for Zeppelin.
+ */
+public class PythonInterpreter extends Interpreter {
+  Logger logger = LoggerFactory.getLogger(PythonInterpreter.class);
+
+  public static final String PYTHON_PATH = "python.path";
+  public static final String DEFAULT_PYTHON_PATH = "/usr/bin/python";
+  private String pythonPath;
+
+  InputStream stdout;
+  OutputStream stdin;
+  BufferedWriter writer;
+  BufferedReader reader;
+  Process process = null;
+  private long pythonPid;
+
+  static {
+    Interpreter.register(
+            "python",
+            "python",
+            PythonInterpreter.class.getName(),
+            new InterpreterPropertyBuilder()
+                    .add(PYTHON_PATH, DEFAULT_PYTHON_PATH,
```

Comment:
actually PYTHONHOME might be better, PYTHONPATH is for modules, so likely we should avoid picking on a similar name to avoid confusion to Python users.
https://docs.python.org/2/using/cmdline.html#envvar-PYTHONHOME
https://docs.python.org/3/using/cmdline.html#envvar-PYTHONHOME